src/MagicScaler/Magic/MatteTransform.cs

```diff
@@ -90,6 +90,8 @@ unsafe private void applyMatteLinearFloat(WICRect prc, float* pixels, int stride
 					ip[1] = ig;
 					ip[2] = ir;
 					ip[3] = 1f;
+
+					ip += Vector<float>.Count;
```

Comment:
Ah, good catch.  This loop is meant to process any pixels that didn't fit into the vector loop, so it will only be entered on odd-width images and only if the CPU supports the AVX2 instruction set.  That's why you get different results on different hardware.

Incrementing the pointer by Vector<float>.Count is incorrect in this case, because with current hardware, that value will be 8 any time this loop is invoked, and the loop only consumes 4 values per iteration.  It will still cause the loop to end, but it's the wrong value and won't work properly on wider vectors when those become available.